### PR TITLE
Feature/project logs

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- #1112

## Changes

- Add project log in
    - Project Create
    - Project Mutations (utils.graphene.middleware::ProjectLogMiddleware)
        - Project Update
        - Project Membership
            - User
            - Usergroup
        - Project Region
        - Project Viz share configuration.
        - Project Join request accept.
        - Project Delete (Soft)
- Add/Update test.
- Admin Panel.
    - Add ProjectChangeLog Model.
    - Add auto complete filters in entry/lead.
Mention related users here if any.

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] flake8 issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/project/admin.py [hunk 1]
```diff
@@ -179,3 +186,30 @@ class ProjectEntryStatsAdmin(admin.ModelAdmin):
...
+    def get_queryset(self, request):
+        return super().get_queryset(request).prefetch_related(
+            'project',
+            'user',
+        )
+
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    @admin.display(description='Diff prettry JSON')
```
Comment (bewakes) [inline apps/project/admin.py]: `prettry` means `pretty`?

File: apps/project/change_log.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+
+from .models import (
+    Project,
+    ProjectChangeLog,
+    ProjectOrganization,
+)
+
+
+def get_flat_dict_diff(list1, list2, fields):
```
Comment (bewakes) [inline apps/project/change_log.py]: Please add argument types wherever possible, especially for utils functions like this.

**Review by bewakes — CHANGES_REQUESTED** [review]

File: apps/project/change_log.py (hunk 2 above)
Comment (thenav56) [reply apps/project/change_log.py]: Done

File: apps/project/admin.py (hunk 1 above)
Comment (thenav56) [reply apps/project/admin.py]: Done

**Review by k9845 — APPROVED** [review]

**Review by bewakes — APPROVED** [review]

**Pull request merged**